src/Features/Core/Portable/Completion/Providers/AbstractDocCommentCompletionProvider.cs

```diff
@@ -1,81 +1,63 @@
-        protected const string CTagName = "c";
-        protected const string CodeTagName = "code";
-        protected const string CommentPrefixTagName = "!--";
-        protected const string CompletionListTagName = "completionlist";
-        protected const string DescriptionTagName = "description";
-        protected const string ExampleTagName = "example";
-        protected const string ExceptionTagName = "exception";
-        protected const string IncludeTagName = "include";
-        protected const string ItemTagName = "item";
-        protected const string ListTagName = "list";
-        protected const string ListHeaderTagName = "listheader";
-        protected const string ParaTagName = "para";
-        protected const string ParamTagName = "param";
-        protected const string ParamRefTagName = "paramref";
-        protected const string PermissionTagName = "permission";
-        protected const string RemarksTagName = "remarks";
-        protected const string ReturnsTagName = "returns";
-        protected const string SeeTagName = "see";
-        protected const string SeeAlsoTagName = "seealso";
-        protected const string SummaryTagName = "summary";
-        protected const string TermTagName = "term";
-        protected const string TypeParamTagName = "typeparam";
-        protected const string TypeParamRefTagName = "typeparamref";
-        protected const string ValueTagName = "value";
-
-        // Attribute names
-        protected const string CrefAttributeName = "cref";
-        protected const string FileAttributeName = "file";
-        protected const string NameAttributeName = "name";
-        protected const string PathAttributeName = "path";
-        protected const string TypeAttributeName = "type";
-
-        private readonly Dictionary<string, string[]> _tagMap =
+        private static readonly ImmutableArray<string> s_alwaysVisibleTagNames = ImmutableArray.Create(SeeElementName, SeeAlsoElementName);
+        private static readonly ImmutableArray<string> s_listTagNames = ImmutableArray.Create(ListHeaderElementName, TermElementName, ItemElementName, DescriptionElementName);
+        private static readonly ImmutableArray<string> s_listHeaderTagNames = ImmutableArray.Create(TermElementName, DescriptionElementName);
+        private static readonly ImmutableArray<string> s_nestedTagNames = ImmutableArray.Create(CElementName, CodeElementName, ParaElementName, ListElementName);
+        private static readonly ImmutableArray<string> s_topLevelRepeatableTagNames = ImmutableArray.Create(ExceptionElementName, IncludeElementName, PermissionElementName);
+        private static readonly ImmutableArray<string> s_topLevelSingleUseTagNames = ImmutableArray.Create(SummaryElementName, RemarksElementName, ExampleElementName, CompletionListElementName);
+
+        private static readonly Dictionary<string, string[]> s_tagMap =
             new Dictionary<string, string[]>
```

Comment:
consider a string-tuple, instead of a string array.  That way we can name things later on, instead of doing things like array[1] + array[2]